### ISSUE TITLE
create() method not working as expected

### ISSUE THREAD

**Issue by jbob** [jhthorsen/mandel#30]
It seems that that the create() method does not honor the defined model. I'm not sure if this is intended behavior or a bug.

The following code works fine, but in my opinion it should produce an error:
```perl
package MyModel;
use Mojo::Base "Mandel";
1;

package MyModel::Person;
use Mandel::Document;
use Types::Standard qw(Int Str);
field name => ( isa => Str );
field age => ( isa => Int );
1;

package main;
my $connection = MyModel->connect("mongodb://localhost/my_db");
my $persons = $connection->collection('person');
my $p1 = $persons->create({ name => 'Bruce', age => "This is not a number", foo => "bar" });
$p1->save;
```
The resulting Object in MongoDB looks like this:
```
{
        "_id" : ObjectId("5d2835e202912b0a06cf544f"),
        "foo" : "bar",
        "name" : "Bruce",
        "age" : "This is not a number"
}
```
Calling `$p1->age('This is also not a number')` later, or `$p1->foo('baz')` produces the expected errors.

**Comment by jhthorsen** [comment]
I don't have time to look at this myself at the moment, but I'm guessing https://github.com/jhthorsen/mandel/blob/master/lib/Mandel/Collection.pm#L36 could be modified to something like this:

```
my ($self, $data) = (@_, {});
my $doc = $self->_new_document(undef, 0);
$doc->$_($data->{$_}) for grep { $doc->can($_) } keys %$data;
return $doc;
```

Or maybe adding a `$doc->validate_fields` method would be better that can be called after the new document object is created.

**Comment by jbob** [comment]
Thanks for confirming, that I didn't miss anything obvious.
I'll see if I can implement one of your suggested fixes myself.

**Comment by jhthorsen** [comment]
Fixed by #31.
